### PR TITLE
Adds support for `dolt log --stat`

### PR DESCRIPTION
**Pull request by stephkyou**
Adds support for `dolt log --stat` which prints the diffstat for each commit in the log.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>77edeea</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>77edeea</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>77edeea</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@stephkyou __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>3a2323a</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>3a2323a</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>3a2323a</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

File: go/cmd/dolt/commands/merge.go [hunk 1]
```diff
@@ -435,11 +440,11 @@ func calculateMergeConflicts(queryist cli.Queryist, sqlCtx *sql.Context, mergeSt
 }
 
 // calculateMergeStats calculates the table operations and row operations that occurred during the merge. Returns a map of
-// table name to MergeStats.
-func calculateMergeStats(queryist cli.Queryist, sqlCtx *sql.Context, mergeStats map[string]*merge.MergeStats, fromRef, toRef string) (map[string]*merge.MergeStats, error) {
+// table name to MergeStats and a bool representing whether all tables were unmodified.
```
Comment (macneale4) [inline go/cmd/dolt/commands/merge.go:443]: bool, set to TRUE when all tables are unmodified.

**Review by macneale4 — APPROVED** [review]

File: go/cmd/dolt/commands/log.go [hunk 2]
```diff
@@ -314,28 +313,152 @@ func logCompact(pager *outputpager.Pager, apr *argparser.ArgParseResults, commit
...
+
+	return nil
 }
 
-func logDefault(pager *outputpager.Pager, apr *argparser.ArgParseResults, commits []CommitInfo) {
+func logDefault(pager *outputpager.Pager, apr *argparser.ArgParseResults, commits []CommitInfo, sqlCtx *sql.Context, queryist cli.Queryist) error {
 	for _, comm := range commits {
 		PrintCommitInfo(pager, apr.GetIntOrDefault(cli.MinParentsFlag, 0), apr.Contains(cli.ParentsFlag), apr.GetValueOrDefault(cli.DecorateFlag, "auto"), &comm)
+		if apr.Contains(cli.StatFlag) {
+			if comm.parentHashes != nil && len(comm.parentHashes) == 1 {
```
[the remark this answers is not in the document]
Comment (stephkyou) [reply go/cmd/dolt/commands/log.go:336]: I can definitely add a comment clarifying why there's a check for 1 parent, from what I can tell git doesn't print stats for merge commits so I don't think there's anything else to do here.

**Comment by coffeegoddd** [comment]
@stephkyou __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>becbce6</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>becbce6</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>becbce6</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

**Pull request merged**